### PR TITLE
[eclipse/xtext#1384] Change to Declarative Pipeline

### PR DESCRIPTION
**Pull request by kthoms**
- add Slack notification
- externalized build call into shell script

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
-	properties([
-		[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', numToKeepStr: '15']]
-	])
-	
-	stage('Checkout') {
-		checkout scm
-	}
-	
-	stage('Build') {
-		sh "./gradlew clean build createLocalMavenRepo -PuseJenkinsSnapshots=true -PJENKINS_URL=$JENKINS_URL -PignoreTestFailures=true --refresh-dependencies --continue"
```
Comment (szarnekow) [inline Jenkinsfile]: Hmm well it was that before, too.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
+        def color = '#00FF00'
+        if (curResult == 'SUCCESS' && currentBuild.previousBuild != null) {
+          curResult = 'FIXED'
+        } else if (curResult == 'UNSTABLE') {
+          color = '#FFFF00'
+        } else if (curResult == 'FAILURE') {
+          color = '#FF0000'
+        }
+        
+        slackSend message: "${curResult}: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}${envName}>", baseUrl: 'https://itemis.slack.com/services/hooks/jenkins-ci/', botUser: true, channel: 'xtext-builds', color: '#00FFFF', token: '1vbkhv8Hwlp3ausuFGj1BdJb'
```
Comment (szarnekow) [inline Jenkinsfile]: Please do not commit tokens.

Comment (szarnekow) [reply Jenkinsfile]: Or URLs

**Comment by cdietrich** [comment]
@kthoms can this PR be closed?

**Comment by cdietrich** [comment]
@kthoms what is the state of this pr?

**Comment by kthoms** [comment]
I will rebase all of them now.

**Pull request merged**